lib/drives/pages/drive_chat_page.dart

```diff
@@ -0,0 +1,158 @@
+import 'package:flutter/material.dart';
+
+import '../../account/models/profile.dart';
+import '../../rides/models/ride.dart';
+import '../../util/trip/pending_ride_card.dart';
+import '../../util/supabase.dart';
+import '../models/drive.dart';
+
+class DriveChatPage extends StatefulWidget {
+  final Drive drive;
+  const DriveChatPage({required this.drive, super.key});
+
+  @override
+  State<DriveChatPage> createState() => _DriveChatPageState();
+}
+
+class _DriveChatPageState extends State<DriveChatPage> {
+  late Drive _drive;
+  late Set<Profile> _riders;
+  late Set<Ride> _pendingRides;
+
+  @override
+  void initState() {
+    super.initState();
+    _drive = widget.drive;
+    _riders = _drive.approvedRides!.map((ride) => ride.rider!).toSet();
+    _pendingRides = _drive.pendingRides!.toSet();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    List<Widget> widgets = [];
+    if (_riders.isEmpty && _pendingRides.isEmpty) {
+      widgets.add(const Center(
+        child: Text("No riders or pending rides"),
+      ));
+    } else {
+      if (_riders.isNotEmpty) {
+        List<Widget> riderColumn = [
+          const Text(
+            "Riders",
+            style: TextStyle(
+              fontSize: 24.0,
+              fontWeight: FontWeight.bold,
+            ),
+          ),
+          const SizedBox(height: 10.0),
+          riderList(),
+        ];
+        widgets.addAll(riderColumn);
+      }
+      widgets.add(const SizedBox(height: 10.0));
+      if (_pendingRides.isNotEmpty) {
+        List<Widget> pendingRidesColumn = [
+          const Text(
+            "Pending Rides",
+            style: TextStyle(
+              fontSize: 24.0,
+              fontWeight: FontWeight.bold,
+            ),
+          ),
+          const SizedBox(height: 10.0),
+        ];
+        pendingRidesColumn.addAll(pendingRidesList());
+        widgets.addAll(pendingRidesColumn);
+      }
+    }
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text('Drive Chat'),
+      ),
+      body: RefreshIndicator(
+        onRefresh: loadDrive,
+        child: ListView.separated(
+          itemCount: widgets.length,
+          itemBuilder: (context, index) {
+            return widgets[index];
+          },
+          separatorBuilder: (BuildContext context, int index) {
+            return const SizedBox(height: 10);
+          },
+        ),
+      ),
+    );
+  }
+
+  List<Widget> pendingRidesList() {
+    List<Widget> pendingRidesColumn = [];
+    if (_pendingRides.isNotEmpty) {
+      pendingRidesColumn = List.generate(
+        _pendingRides.length,
+        (index) => PendingRideCard(
+          _pendingRides.elementAt(index),
+          reloadPage: loadDrive,
+          drive: _drive,
+        ),
+      );
+    }
+    return pendingRidesColumn;
+  }
+
+  Row profileRow(Profile profile) {
```

Comment:
note: this will have to be changed when we merge the account page PR

Reply:
yes but that should be changed anyways when we do the chat

Reply:
account page is merged, let's maybe this change this anyways in case we don't get to the chat for the next few weeks